### PR TITLE
Power iteration for convolution

### PR DESCRIPTION
**Pull request by cofri**
The power iteration algorithm is introduced to compute the maximum singular value of a convolution. The function `spectral_normalization_conv()` returns a normalized convolution kernel.

### CODE REVIEW

**Review by franckma31 — APPROVED** [review]
Fine

File: deel/lip/normalizers.py [hunk 1]
```diff
@@ -238,3 +240,155 @@ def spectral_normalization(
...
+        if pad_func is None:
+            return tf.nn.conv2d_transpose(u, w, output_shape, stride, pad_type)
+        else:
+            u_upscale = _zero_upscale2D(u, (stride, stride))
+            w_adj = _maybe_transpose_kernel(w, True)
+            return _conv(u_upscale, w_adj, stride=1)
+
+    def body(_u, _v, _old_u, _norm_u):
+        _old_u = _u
+        u = _u  # normalization should be done before
```
Comment (thib-s) [inline deel/lip/normalizers.py:299]: seem like the normalization should be done here instead of l312

**Review by thib-s — CHANGES_REQUESTED** [review]
it seems like a the normalization isn't done a the right place, otherwise LGTM

File: deel/lip/normalizers.py (hunk 1 above)
Comment (cofri) [reply deel/lip/normalizers.py]: I've changed the comment to be more understandable

**Review by franckma31 — APPROVED** [review]
fine

**Review by thib-s — APPROVED** [review]
Look good to me!
